lib/api/pageable_api_resource.js

```diff
@@ -8,15 +8,22 @@ class PageableApiResource extends ApiResource {
    *                             'paging'. The second property is holds the list of content (videos, channels, etc.).
    *                             This is the name of that second property.
    * @param {{href: {next},next}} paging
+   * @param {string} pagingProp - Sometimes paginated metadata comes in a different field like for the Analytics API
    */
-  constructor (context, dataProp, data, paging) {
+  constructor (context, dataProp, data, paging, pagingProp = 'paging') {
     super(context)
 
     this.data = data
     this.dataProp = dataProp
+    this.pagingProp = pagingProp
 
-    if (paging.hasOwnProperty('next')) {
-      this.nextPageUrl = paging.next.hasOwnProperty('href') ? paging.next.href : paging.next
+    if (paging.next) {
+      this.nextPageUrl = paging.next.href ? paging.next.href : paging.next
+
+      // Fixing paginated urls not coming with the initial API endpoint
```

Comment:
Not sure what you mean here. Could your give an example?

Reply:
Their API looks not yet uniform, the Analytic pagination is very different than the Channel API one:
- The next URL is null when there's no more next data
- The paging field name is different, it's pagination for Analytics
- The next/previous/actual URLs returned don't have the root domain in the endpoint, like **/v1/xxxxxx/xxxx.json**